server/ServerFSM.cpp

```diff
@@ -735,6 +741,20 @@ sc::result Idle::react(const Hostless&) {
     std::vector<PlayerSaveGameData> player_save_game_data;
     server.InitializePython();
     server.LoadChatHistory();
+
+    if (load_or_quickstart) {
+        std::string subdir = server.m_galaxy_setup_data.GetGameUID();
+        boost::filesystem::path autosave_dir_path = GetServerSaveDir() / (subdir.empty() ? "auto" : subdir);
+        if (IsExistingDir(autosave_dir_path)) {
+            auto saves = ListDir(autosave_dir_path, IsAutoSaveFile);
+            for (const auto& save : saves) {
+                if (PathToString(save) > autostart_load_filename) {
```

Comment:
maybe add a comment about why this ordering of autosave filenames gets the most recent one?

Reply:
Added.